File: python/80 Library.py
class LibrartItem:
    def __init__(self,itemId,title,author):
        self.itemId=itemId
        self.title=title
        self.author=author
        self.isCheckedOut = False

    def checkOut(self):
        if not self.isCheckedOut:
            self.isCheckedOut = True
            print(f"'{self.title}' (ID: {self.itemId}) has been checked out.")
        else:
            print(f"'{self.title}' (ID: {self.itemId}) is already checked out.")

    def returnItem(self):
        if self.isCheckedOut:
            self.isCheckedOut = False
            print(f"'{self.title}' (ID: {self.itemId}) has been returned.")
        else:
            print(f"'{self.title}' (ID: {self.itemId}) was not checked out.")
    
class DVD(LibrartItem):
    def __init__(self, itemId, title, author,duration):
        super().__init__(itemId, title, author)
        self.duration=duration

class Magazine(LibrartItem):
    def __init__(self, itemId, title, author,issueNum):
        super().__init__(itemId, title, author)
        self.issueNum=issueNum

class Book(LibrartItem):
    def __init__(self, itemId, title, author,isbnNum):
        super().__init__(itemId, title, author)
        self.isbnNum=isbnNum

b1 = Book("B001", "The Great Gatsby", "F. Scott Fitzgerald", "978-0743273565")
m1 = Magazine("M001", "National Geographic", "Various", "July 2025")
dvd1 = DVD("D001", "Inception", "Christopher Nolan", 148)

b1.checkOut()
b1.checkOut()
m1.checkOut()
dvd1.returnItem()

print(f"Book '{b1.title}' checked out: {b1.isCheckedOut}")
print(f"Magazine '{m1.title}' checked out: {m1.isCheckedOut}")
print(f"DVD '{dvd1.title}' checked out: {dvd1.isCheckedOut}")

b1.returnItem()
m1.returnItem()

print(f"Book '{b1.title}' checked out: {b1.isCheckedOut}")
print(f"Magazine '{m1.title}' checked out: {m1.isCheckedOut}")

File: java/80 Library.java
package code;
class LibraryItem{
	String itemID,title,author;
	boolean isCheckOut;

	public LibraryItem(String itemID, String title, String author) {
		this.itemID = itemID;
		this.title = title;
		this.author = author;
		this.isCheckOut=false;
	}
	void checkOut() {
		if(!this.isCheckOut) {
			this.isCheckOut=true;
			System.out.println("Checked out");
		}
		else {
			System.out.println("Already checked out");
		}
	}
	void returnItem() {
		if(this.isCheckOut) {
			this.isCheckOut=false;
			System.out.println("Returned");
		}
		else {
			System.out.println("Not checked out");
		}
	}
}
class Book extends LibraryItem{
	int ISBN;

	public Book(String itemID, String title, String author, int iSBN) {
		super(itemID, title, author);
		ISBN = iSBN;
	}
	
}
class DVD extends LibraryItem{
	int duration;

	public DVD(String itemID, String title, String author, int duration) {
		super(itemID, title, author);
		this.duration = duration;
	}
}
class Magazine extends LibraryItem{
	int issuenum;

	public Magazine(String itemID, String title, String author, int issuenum) {
		super(itemID, title, author);
		this.issuenum = issuenum;
	}
}
public class Test {
	public static void main(String[] args) {
		Book book1 = new Book("B001", "The Java Handbook", "John Doe", 978123456);
        DVD dvd1 = new DVD("D001", "Learning Java in 10 Hours", "Jane Smith", 600);
        Magazine mag1 = new Magazine("M001", "Tech Monthly", "Various", 105);

        book1.checkOut();
        book1.checkOut();

        dvd1.returnItem();
        dvd1.checkOut();
        dvd1.returnItem();
	}
}